packages/material-ui-lab/src/Slider/Slider.js

```diff
@@ -306,6 +308,24 @@ class Slider extends React.Component {
     });
   };
 
+  handleMouseEnter = event => {
+    // If the slider was being interacted with but the mouse went off the window
+    // and then re-entered while unclicked then end the interaction.
+    if (event.buttons === 0) {
+      this.handleDragEnd(event);
+    }
+  };
+
+  handleMouseLeave = event => {
+    // If the mouse moves off the window then move the slider to the extremity in the
+    // direction of the window edge.
+    const { min, max, vertical } = this.props;
+    const percent = calculatePercent(this.containerRef, event, vertical, this.isReverted());
+    const value = percentToValue(percent, min, max);
+
+    this.emitChange(event, value);
```

Comment:
Seems like it's not adressing the sticky issue but some use case. What's the motivation behind this?

If I leave the window vertically then the slider jumps horizontally. If I leave horizontally on a vertical slider then it jumps vertically. Changing the value when I leave the window doesn't sound like it matches any intention especially since it jumps back again if I enter the window again at the positon I left.

Reply:
Overall this PR addresses some of the quirks I noticed with the slider when mousing off the window. The `mouseenter` event handler addresses the stickiness issue while this change here addresses this bullet:

> - When mousing off the window the slider is moved to the extreme end instead of staying at the last-seen position of the mouse.

In master, when moving the mouse quickly off the window the value can get stuck at a position in the middle of the slider, which is not what I would expect. For example, if you drag off the left side of the screen it might get stuck at a value of `5` even though the left extreme is `0`.

> it jumps back again if I enter the window again at the positon I left.

If the mouse is released outside the window then it won't jump to zero even though a quick slide to the left would imply a desire to move the slider all the way down.

This behavior is something that can be achieved by the containing application by adding the same logic that's been added here but I figured I'd propose it in the interest of keeping the components behaving as cleanly as possible. This behavior can be replicated on the material UI docs website when laid out for a narrow window, as well (re-added cursor manually because print screen on windows didn't include it):

![image](https://user-images.githubusercontent.com/734200/48017529-569b0c80-e0e3-11e8-83db-d0ee74a9a51f.png)

If it doesn't belong in the component or needs more comments or changes let me know I can make the updates!

Reply:
Comparing the Netlify preview of this PR to the live docs, it does appear to fix the issues raised, as well as mimicking the behavior of the native range input slider.

Reply:
So seems like I looked at an older version 

>When mousing off the window the slider is moved to the extreme end instead of staying at the last-seen position of the mouse.

Is not happening anymore which is what I would expect.

Reply:
@gkjohnson But the whole `handleMouseLeave` is now basically a no-op. Why not just remove it? At least the comments are outdated.

Reply:
@eps1lon 

> But the whole handleMouseLeave is now basically a no-op. Why not just remove it? At least the comments are outdated.

Can you elaborate on what you mean? In what way is it a no-op? Maybe I misunderstood your last comment.

Thanks!

Reply:
There is no observable value change when mousing off the window. It can be at most 1px.

Reply:
When moving the mouse quickly it can move far more than 1px between move and leave events and a move will not necessarily be fired right before mouse leave is called, which means significant movement could have happened without a corresponding event firing. The following code:

```js
document.body.addEventListener('mousemove', e => console.log('move', e.pageX));
document.body.addEventListener('mouseleave', e => console.log('leave', e.pageX));
```

Can yield the following statements when moving the mouse out of the window:

```
...
move 468
move 369
move 258
move 137
move 63
leave -2
```

Hopefully I'm understanding correctly! I ran the above in both the latest Firefox and Chrome.

Reply:
Well maybe. The comment is still wrong. It's not being moved the its extreme values and it shouldn't be.

Reply:
That's fair -- the function is not explicitly moving the slider to an end. Looking at the function again it could just call `handleMouseMove` directly. Would that make it more clear? For example:

```js
 handleMouseLeave = event => {
    // The mouse will have moved between the last mouse move event
    // this mouse leave event
    this.handleMouseMove(event);
}
```

Reply:
@gkjohnson Perfect!